client/state/analytics/actions/enhance-with-site-main-product.js

```diff
@@ -0,0 +1,46 @@
+import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import { ANALYTICS_PAGE_VIEW_RECORD } from 'calypso/state/action-types';
+import isDomainOnly from 'calypso/state/selectors/is-domain-only-site';
+import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+/**
+ * Enhances any Redux action that denotes the recording of a page view analytics event with an additional property
+ * to specifies the main product of this site:
+ * `domain` - domain only, without email subscription
+ * `email` - domain only with email subscription
+ * `site` - regular site
+ *
+ * @param {object} action - Redux action as a plain object
+ * @param {Function} getState - Redux function that can be used to retrieve the current state tree
+ * @returns {import('redux').AnyAction} the new Redux action
+ * @see client/state/utils/withEnhancers
+ */
+export function enhanceWithSiteMainProduct( action, getState ) {
+	if ( action.type !== ANALYTICS_PAGE_VIEW_RECORD ) {
+		return action;
+	}
+
+	const state = getState();
+	const siteId = getSelectedSiteId( state );
+	let mainProduct = 'site';
+
+	if ( isDomainOnly( state, siteId ) ) {
+		mainProduct = 'domain';
+
+		// Domain only site should only have one single domain non-wpcom domain.
+		const nonWPCOMDomain = getDomainsBySiteId( state, siteId ).find(
+			( domain ) => ! domain.isWPCOMDomain
+		);
+
+		if ( hasTitanMailWithUs( nonWPCOMDomain ) || hasGSuiteWithUs( nonWPCOMDomain ) ) {
```

Comment:
Should we handle the case when `nonWPCOMDomain` is `undefined`? Could that even happen if we're certain we're working with a domain-only site?

Reply:
This really shouldn't happen, even if it does, the `hasTitanMailWithUs()` and `hasGSuiteWithUs()` will return `false`, and the `site_main_product` property will be set to `domain`, which I think is the correct behaviour, given that it's a domain-only site. 

I've also tested deleting the domain for a domain-only site, in which case, users lose access to the domain-site altogether. So I don't think we'll ever end up with a undefined `nonWPCOMDomain` unless for some really random error.